### PR TITLE
Ktlint plugin

### PR DESCRIPTION
**Pull request by mckelveygreg**
So this uses a ktlint plugin that takes care of all of the configuration. [There is a way to overwrite](https://github.com/jlleitschuh/ktlint-gradle#configuration) it if we would like.

The readme step sets you up to add it as a `pre-commit` step, but I feel like there is an easier way to do that?? @Jamdan2

### CODE REVIEW

**Comment by will-hou** [comment]
This looks wonderful, thanks! @codingvic took a look at this today and should update his branch on Thursday with the changes

File: README.md [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+
+## Project setup
+After gradle has synced, either use the gradle menu to run
+> Tasks > help > addKtlintFormatGitPreCommitHook
+
+or in your commandline, run
+
+```
+chmod +x gradlew
+./gradlew addKtlintFormatGitPreCommitHook
```
Comment (ktsukamoto01) [inline README.md:12]: Is this in the terminal? Because it doesn't work when I run it there.

Comment (mckelveygreg) [reply README.md:12]: Did you try running this with gradle, like the built in gradle thing in Intellij?

**Review by will-hou — APPROVED** [review]
Code has been tested on the robot without anything breaking!

**Comment by mckelveygreg** [comment]
Well lets get this merged and pulled into some of the working branches for some code quality!! 
@Jamdan2 May this have your divine approval? 👼

File: README.md (hunk 1 above)
Comment (gnawinganimal) [reply README.md:12]: I'm pretty sure this only works on linux. On windows just run 
`./gradlew adKtlintFormatGitPreCommitHook`

Comment (gnawinganimal) [reply README.md:12]: If you are using the native windows terminal, omit the `./`

**Review by gnawinganimal — COMMENTED** [review]
lgtm

**Pull request merged**